src/DynamoCoreWpf/Views/Search/LibrarySearchView.xaml.cs

```diff
@@ -682,7 +682,8 @@ private void OnTopResultListBoxIcgStatusChanged(object sender, EventArgs e)
                 topResultListBox.ItemContainerGenerator.StatusChanged -= OnTopResultListBoxIcgStatusChanged;
                 Dispatcher.BeginInvoke(new Action(() =>
                     {
-                        UpdateHighlightedItem(GetListItemByIndex(topResultListBox, 0));
+                        if (viewModel.CurrentMode == SearchViewModel.ViewMode.LibrarySearchView)
+                            UpdateHighlightedItem(GetListItemByIndex(topResultListBox, 0));
```

Comment:
Hi @aosyatnik, it would have been much better if we don't dispatch the task if the mode isn't right:

```csharp
if (viewModel.CurrentMode == SearchViewModel.ViewMode.LibrarySearchView)
{
    Dispatcher.BeginInvoke(new Action(() =>
    {
        UpdateHighlightedItem(GetListItemByIndex(topResultListBox, 0));
    }), DispatcherPriority.Input);
}
```